scripts/webidl/convert.js

```diff
@@ -0,0 +1,44 @@
+"use strict";
+const fs = require("fs");
+const path = require("path");
+
+const recursive = require("recursive-readdir");
+const webidl2js = require("webidl2js");
+
+const optimist = require("optimist")
+  .usage("Pull all webidl segments froma a web page")
+  .string("input")
+  .alias("input", "i")
+  .describe("input", "the input file or directory")
+  .string("implDir")
+  .alias("implDir", "m")
+  .describe("implDir", "the directory of the implementation files (relative to the output dir)")
+  .string("outputDir")
+  .alias("outputDir", "o")
+  .describe("outputDir", "output directory")
+  .alias("h", "help")
+  .describe("h", "show the help");
+
+if (optimist.argv.help) {
+  optimist.showHelp();
+  process.exit(0);
+}
+
+let inputFiles = [optimist.argv.input];
+
+const inputStat = fs.statSync(optimist.argv.input);
+if (inputStat.isDirectory()) {
+  recursive(optimist.argv.input, readFiles);
+} else {
+  readFiles(null, [optimist.argv.input]);
+}
+
+function readFiles(err, files) {
+  let src = "";
+  files.forEach(function (f) {
+    src += fs.readFileSync(f, { encoding: "utf8" });
+  });
+
+  webidl2js.generate(src, optimist.argv.outputDir, optimist.argv.implDir, { suppressErrors: true });
```

Comment:
I am a bit confused what the input to and output of webidl2js.generate is. It takes a string consisting of ... the concatenation of all IDL in the currently-being-recursed directory? But all calls use the same output and impl directory?

What errors are suppressed?